### PR TITLE
Add more Integrationtests

### PR DESCRIPTION
**Pull request by rndmh3ro**
* the update-script creates all tests. It does this by:
  * create creation-tests from the examples in the plugin-files
  * create deletion-tests
  * create tests with a wrong password (to follow a error path) - these failing tasks get ignored and then a assert-task runs that checks the result of the failed task
  * create tests with a wrong icinga-host (to follow another error path) - these failing tasks get ignored and then a assert-task runs that checks the result of the failed task

I'll probably need some documentation for the test creation and script, but I want to get this out here first to let you review.

### CODE REVIEW

File: tests/integration/targets/icinga/tasks/main.yml [hunk 1]
```diff
@@ -16,6 +17,71 @@
...
+    - ./icinga_hostgroup.yml
+    - ./icinga_servicegroup.yml
+    - ./icinga_service_template.yml
+    - ./icinga_service_apply.yml
+    - ./icinga_hostgroup.yml
+    - ./icinga_host_template.yml
+    - ./icinga_host.yml
+
+    ### failing tests
+    - ./wrong_host_icinga_command_template.yml
```
Comment (schurzi) [inline tests/integration/targets/icinga/tasks/main.yml:37]: this seems very redundant, why not change a variable and then rerun the tests? (same for password)

**Comment by schurzi** [comment]
why did the coverage for all the other modules not increase? It seems everything should now be at almost 100% but only `icinga_service_apply` got improvements

File: tests/integration/targets/icinga/tasks/main.yml (hunk 1 above)
Comment (rndmh3ro) [reply tests/integration/targets/icinga/tasks/main.yml:37]: I'm definetly interested in improving this. However I don't want to have to run `ansible-test` two times or build some kind of script around this. Just running ansible-test should be enough.

**Comment by rndmh3ro** [comment]
> why did the coverage for all the other modules not increase? It seems everything should now be at almost 100% but only `icinga_service_apply` got improvements

🤷‍♀️

File: tests/integration/targets/icinga/tasks/main.yml (hunk 1 above)
Comment (schurzi) [reply tests/integration/targets/icinga/tasks/main.yml:37]: hmm, is this file a standard playbook? so we could split this into different stages where the right variables are set by `set_fact`?

Comment (rndmh3ro) [reply tests/integration/targets/icinga/tasks/main.yml:37]: Good idea, I'll try it.

Comment (rndmh3ro) [reply tests/integration/targets/icinga/tasks/main.yml]: Since we still need the assert-tasks on every task, we cannot simply change the variable.

**Comment by rndmh3ro** [comment]
@schurzi, I'll keep the PR like that for now.
Additional coverage can later be added.

**Review by schurzi — APPROVED** [review]

**Pull request merged**